master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -851,55 +851,74 @@ private[celeborn] class Master(
     statusSystem.handleRemoveWorkersUnavailableInfo(unavailableWorkers, requestId)
   }
 
-  private def computeUserResourceConsumption(userIdentifier: UserIdentifier)
-      : ResourceConsumption = {
-    val current = System.currentTimeMillis()
-    if (userResourceConsumptions.containsKey(userIdentifier)) {
-      val resourceConsumptionAndUpdateTime = userResourceConsumptions.get(userIdentifier)
-      if (current - resourceConsumptionAndUpdateTime._2 > masterResourceConsumptionInterval) {
-        val newResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
-          workerInfo.userResourceConsumption.asScala.get(userIdentifier)
-        }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
-        userResourceConsumptions.put(userIdentifier, (newResourceConsumption, current))
-        newResourceConsumption
-      } else {
-        resourceConsumptionAndUpdateTime._1
-      }
-    } else {
-      val newResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
-        workerInfo.userResourceConsumption.asScala.get(userIdentifier)
-      }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
-      userResourceConsumptions.put(userIdentifier, (newResourceConsumption, current))
-      newResourceConsumption
+  private def handleResourceConsumption(userIdentifier: UserIdentifier): ResourceConsumption = {
+    val userResourceConsumption = computeUserResourceConsumption(userIdentifier)
+    gaugeResourceConsumption(userIdentifier)
+    val subResourceConsumptions = userResourceConsumption.subResourceConsumptions
+    if (CollectionUtils.isNotEmpty(subResourceConsumptions)) {
+      subResourceConsumptions.asScala.keys.foreach { gaugeResourceConsumption(userIdentifier, _) }
     }
+    userResourceConsumption
   }
 
-  private def handleCheckQuota(
+  private def gaugeResourceConsumption(
       userIdentifier: UserIdentifier,
-      context: RpcCallContext): Unit = {
-
+      applicationId: String = null): Unit = {
+    val resourceConsumptionLabel =
+      if (applicationId == null) userIdentifier.toMap
```

Comment:
IMO we should clean these metrics regularly, such as when the application timeout?